### PR TITLE
Performance Wizard UI Improvements

### LINKED ISSUE #180: Adjust Performance wizard screens
**Is your feature request related to a problem? Please describe.**
This is related to #141 Performance Wizard. 
We got some good feedback from the Muriel team about the first screen. There is a lot of information on it, it's not structured great, and the informational copy style is currently implemented with a lighter grey text color which questions the emphasis and importance of the information. Basically this is a good opportunity to rethink this screen.

Current implementation:
screencapture-newstest-ngrok-io-wp-admin-admin-php-2019-07-26-12_25_18

**Suggested updates:**
Based on the feedback and the updated management UI we're implementing, I'm proposing the following updates?
- Update title and subtitle on all screens screen (for better context).
- Implement the tabs to separate the AMP information and the Advanced PWA features
- Update copy and text structure on AMP tab screen
- Add sub nav for Advanced PWA features to access the three screens
- Adjust button text Advanced PWA screens from "Continue" to "Save" (if from a technical point we don't need it because it auto saves, except probably Push Notifications, we might not need a button at all)

**Updated designs ([in Figma](https://www.figma.com/file/CFxMAXFD0DxNFcVkDkQw4gLb/Newspack-Plugin-v1?node-id=1039%3A1556)):**

Screenshot 2019-07-26 12 30 45

_Add to Home screen conditionally shows an image uploader when enabled for adding a favicon:_
Screenshot 2019-07-26 12 35 10

_Offline usage has a simple on/off toggle:_
Screenshot 2019-07-26 12 35 40

_Push Notifications have conditional fields that show only when this feature is enabled:_
Screenshot 2019-07-26 12 36 45

### PR DESCRIPTION
**Pull request by jeffersonrabb**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This updates the UI for the Performance Wizard based on feedback in https://github.com/Automattic/newspack-plugin/issues/180. This is the first wizard with both tabbed and secondary navigation. As it stands currently, Offline Usage features are always going to be on, so I removed the toggle and left the description only. 

<details>
Screen Shot 2019-07-29 at 7 17 22 AM

Screen Shot 2019-07-29 at 7 17 27 AM

Screen Shot 2019-07-29 at 7 17 29 AM

Screen Shot 2019-07-29 at 7 17 36 AM

Screen Shot 2019-07-29 at 7 17 38 AM

</details>

Blocked by:

https://github.com/Automattic/newspack-plugin/pull/217
https://github.com/Automattic/newspack-plugin/pull/216
https://github.com/Automattic/newspack-plugin/pull/215

Closes #180.

### How to test the changes in this Pull Request:

1. `npm run build:webpack`
2. Navigate to the Performance Wizard
3. Compare to screenshots above and visual feedback in https://github.com/Automattic/newspack-plugin/issues/180.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/performance/index.js [hunk 1]
```diff
@@ -89,6 +92,36 @@ class PerformanceWizard extends Component {
 	render() {
 		const { pluginRequirements } = this.props;
 		const { settings } = this.state;
+		const tabbedNavigation = [
+			{
+				label: __( 'AMP' ),
```
[the remark this answers is not in the document]
Comment (jeffersonrabb) [reply assets/wizards/performance/index.js]: I'm actually not sure? I had thought this _was_ meant to describe the performance gains brought about by AMP (in which case the text will need to be reworked a bit) but you may be right? @sonjaleix?

File: assets/wizards/performance/views/add-to-homescreen/index.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ class AddToHomeScreen extends Component {
 		const { updateSetting, settings } = this.props;
 		return (
 			<Fragment>
+				<h4>{ __( 'Encourage your users to add your news site to their homescreen. ' ) }</h4>
```
[the remark this answers is not in the document]
Comment (jeffersonrabb) [reply assets/wizards/performance/views/add-to-homescreen/index.js]: I standardized all as `homescreen` in https://github.com/Automattic/newspack-plugin/pull/219/commits/5d20b0765938482dda05b2eb5373fd3accdbd4fa

**Comment by jeffersonrabb** [comment]
> Offline usage screen is missing the toggle [critical, needs fixing]

I threw a note about this in the header. As it stands, offline usage is always on in the Progressive WP, so there is actually no way to fully turn it off. I removed the toggle because it seemed misleading considering this.

**Comment by sonjaleix** [comment]
> I threw a note about this in the header. As it stands, offline usage is always on in the Progressive WP, so there is actually no way to fully turn it off. I removed the toggle because it seemed misleading considering this.
The question is then if we should show it at all. Something @rickybanister might want to explore.

**Comment by jeffersonrabb** [comment]
> Push notifications server key and sender id fields need error handling. They need to be required fields if the feature is toggled on.

This does happen currently. However, because that screen is fairly tall, it's likely you don't see the notification because you've scrolled down a bit and the notification appears at the very top of the screen. First of all, could you verify that this is true? And second, this may mean we should reconsider how we display error messages—if so, probably the work of a separate PR.

**Comment by rickybanister** [comment]
Offline usage being 'always on' makes sense. The thing I don't fully understand from what I see here is how I'd turn off PWA stuff entirely without disabling Newspack, is that possible? Should it be?

**Comment by jeffersonrabb** [comment]
> how I'd turn off PWA stuff entirely without disabling Newspack

Each Wizard has certain plugin dependencies. When the Wizard loads, if these dependencies have not been installed the user is presented with a list and a single button that will install them all. These are distinct from the "required plugins" which are installed when a user first sets up Newspack (Jetpack, Site Kit, AMP, etc).

In the case of the Performance Wizard, the plugin is "Progressive WP". If the user _really_ wanted to disable it, they'd need to deactivate the plugin manually and then not proceed past the first screen of the Performance Wizard. The main question is why would they want to?

**Comment by jeffersonrabb** [comment]
> Offline usage being 'always on' makes sense. 

Does this mean we should remove the screen entirely? Or keep it in current state as a purely informational screen with no actionable UI?

**Comment by rickybanister** [comment]
Informational screen makes perfect sense, if a feature is hidden and falls in the woods does it make a sound?

I think the rest of that makes sense, I'm just remembering all the back and forth about always-on modules in Jetpack and how people wanted control even if they didn't use it or it didn't fully make sense. No need to complicate things though. If there's a way to turn things off for debugging, then that's sufficient.

**Comment by sonjaleix** [comment]
> First of all, could you verify that this is true? And second, this may mean we should reconsider how we display error messages—if so, probably the work of a separate PR.
Yes, confirmed, the error message shows at the top of the screen. This is not ideal because it can easily be missed and the user can feel at a loss. Contextual error messages that display closest in proximity to where the error occurs are the most helpful and user friendly. Agreed, we'd want to open a separate PR and implement a solution for this anywhere we have required fields.

**Comment by jeffersonrabb** [comment]
Taking stock of this one, here's what I believe needs to be resolved before it can land:

- Resolve questions about the first navigation tab. Is it meant to be a description of the performance benefits of AMP that are baked in, or a general description of PWA features.
- If we're OK having an offline usage screen that is purely informational, is the current text sufficient?

Questions that should be filed for future work:

- Rethink error messaging approach, since messages at the top of the page may be lost above the fold.
- Examine SAVE button usage on Wizard screens. In which circumstances should changes be written to the database immediately vs. when user explicitly clicks SAVE.
- General question raised elsewhere about the Plugin Installer UI. Do we want to install silently? Do we want to offer a way to disable a plugin through Newspack? Etc.

cc: @rickybanister

**Comment by rickybanister** [comment]
> Rethink error messaging approach, since messages at the top of the page may be lost above the fold.

I'm not familiar of a Muriel pattern or standard here, but in lieu of that we should look to Calypso settings. There we post a banner at the top of the screen (or a sticky global floating one) and also provide in-context error messages. This helps catch both circumstances (being scrolled away or only seeing the global message and wondering where the problem is). 

>Examine SAVE button usage on Wizard screens. In which circumstances should changes be written to the database immediately vs. when user explicitly clicks SAVE.

Also looking to Calypso and past work here—we always saved when a toggle was modified automatically, but relied on a save button when a form needed to be submitted. Obviously it wouldn't make sense to make front-end changes to a site based on an on-change event, so that's when we'd ask folks to click a save button. I'm a little dubious that people will find the save button when it's outside of the card (though that might be the muriel pattern) so let me check on that. We should definitely make sure we're handling the case where someone tries to exit the page with unsaved changes (showing a browser alert or similar). 

>General question raised elsewhere about the Plugin Installer UI. Do we want to install silently? Do we want to offer a way to disable a plugin through Newspack? Etc.

Since we're obfuscating or providing alternate UI for other plugins I think we should philosophically act like plugins===features when it makes sense. I don't think we need to hide the installation process, but rather speak about it as a means to an end ("To set up X we'll install Z for you and then you'll be able to configure W"). I said elsewhere, for hosted customers we can probably hide it more, but what we want to avoid is someone missing that we installed a bunch of necessary things, then they visit their plugins page and see strange plugins, deactivate something and break functionality.

**Comment by jeffersonrabb** [comment]
Superseded by https://github.com/Automattic/newspack-plugin/pull/253, closing.

**Pull request closed without merging**